components/homme/src/theta-l_kokkos/prim_driver_mod.F90

```diff
@@ -422,14 +422,16 @@ subroutine prim_run_subcycle(elem, hybrid, nets, nete, dt, single_column, tl, hv
 
     call TimeLevel_Qdp(tl, dt_tracer_factor, n0_qdp, np1_qdp)
 
+#if !defined(CAM) && !defined(SCREAM)
```

Comment:
I am going to clean up this file in the next forcing PR. Why did you have to touch it? This particular case, when not SCREAM, is handled in `subroutine init_logic_for_push_to_c()` .

Can you not touch this file in this PR?

Reply:
I needd to ifdef out the next few lines, since `test_mod` is only used if CAM and SCREAM are not defined. When building this file in SCREAM, it caused a link error.

Reply:
I want to hide all logic about push/pull etc. in `init_logic_for_push_to_c` and `init_logic_for_push_to_f`, cause it is cleaner. We could try to have a dummy function that would call compute_test_forcing if !SCREAM? it is ugly though.

Reply:
I'm ok hiding all the logic, and reducing the number places where we use preprocessor macros. Unfortunately, all the stuff that is undefined must be ifdeffed out.

We can let SCREAM compiler all the forcing stuff, and simply rely on the runtime switches set in the init routines for skipping that stuff. Either way, we either ifdef out code in all places, or we never  ifdef it out. The former leads to arguably cleaner code (less ifdefs, all logic confined to the init routines), while the latter allows to not compile all the test stuff for SCREAM use (but we are currently compiling it anyways, so maybe it's fine).

Reply:
Actually, this is necessary:

```
#if !defined(CAM) && !defined(SCREAM)
    ! Test forcing is only for standalone Homme (and only for some tests/configurations)
    if (compute_forcing_and_push_to_c) then
```
since EAM does not compile all the test_mod stuff, so these lines cannot be compiled for CAM builds. SCREAM does build all the test_mod stuff, so it would be ok for SCREAM, but it definitely would not compile in EAM without the `#if !defined(CAM)..` line.

Reply:
Since no EAM compset uses theta-l_kokkos, this might be a somewhat pointless issue. But in the long term, it must be addressed.

Reply:
Long story short, we have 2 approaches:

- ifdef out all occurrences of compute/push forcing
- do not ifdef out any occurrence, but make sure test_mod and test_src/* files are always built.

Reply:
Do you need changes in this file for this PR? is there a test case in scream already?

I can integrate this as is, but I will change this file again later.

Reply:
@oksanaguba I hypothesize that if the test SMS_B.ne4_ne4.F-EAM-AQP1.machine_compiler.eam-hommexx is modified to build with theta-l_kokkos instead of preqx_kokkos, this linking error will be evident in the E3SM repo without needing to look at the SCREAM one.